brion/csvConfig.cpp

```diff
@@ -0,0 +1,145 @@
+/* Copyright (c) 2013-2018, EPFL/Blue Brain Project
+ *                          Jonas Karlsson <jonas.karlsson@epfl.ch>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "csvConfig.h"
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <regex>
+#include <streambuf>
+#include <string>
+#include <unordered_map>
+
+namespace
+{
+// Taken from:
+// https://stackoverflow.com/questions/890164/how-can-i-split-a-string-by-a-delimiter-into-an-array
+std::vector<std::string> explode(const std::string& str, const char& ch)
```

Comment:
Use this, https://theboostcpplibraries.com/boost.tokenizer I think you can support multiple spaces this way.

Another simple approach is to use a std::stringstream and use << to extract strings as it skips white space by default.